packages/backend/server/src/plugins/copilot/context/job.ts

```diff
@@ -0,0 +1,176 @@
+import { randomUUID } from 'node:crypto';
+
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import { Prisma, PrismaClient } from '@prisma/client';
+import OpenAI from 'openai';
+
+import {
+  AFFiNELogger,
+  Config,
+  EventBus,
+  JobQueue,
+  metrics,
+  OnEvent,
+  OnJob,
+} from '../../../base';
+import { DocReader } from '../../../core/doc';
+import { OpenAIEmbeddingClient } from './embedding';
+import { Chunk, Embedding, EmbeddingClient } from './types';
+import { checkEmbeddingAvailable } from './utils';
+
+declare global {
+  interface Jobs {
+    'doc.embedPendingDocs': {
+      workspaceId: string;
+      docId: string;
+    };
+
+    'doc.embedPendingFiles': {
+      contextId: string;
+      fileId: string;
+      chunks: Chunk[];
```

Comment:
job system built on redis, do not use binary data as job payload

Reply:
the maximum size of each group of chunks is now within 256k

Reply:
use file id + chunk: { start, end } instead of put content in payload

Reply:
or embed a file in a single job, even this is far better than delivering by content chunk